### PR TITLE
Add typedef test

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
Add tests to verify #1327

### CODE REVIEW

**Comment by Fank** [comment]
:+1: for adding this to ensure such syntax is also valid

File: test/rules/typedef/call-signature/test.ts.lint [hunk 1]
```diff
@@ -23,4 +23,17 @@ setTimeout(function() {}, 1000);
...
+        this._foo.test({
+            callback: () => {
+                // asd
+            },
+        });
+    }
+}
+
+function foo(bar: any) :void {}
+foo({bar: ()=>{}});
```
Comment (adidahiya) [inline test/rules/typedef/call-signature/test.ts.lint]: Minor: EOF newline

File: test/rules/typedef/arrow-call-signature&parameter/test.ts.lint [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class AComponent {
```
Comment (adidahiya) [inline test/rules/typedef/arrow-call-signature&parameter/test.ts.lint]: I'm hesitant to introduce a "&" character into the folder path (call me paranoid) -- can you rename the folder to "arrow-call-signature-and-parameter"?

**Comment by YuichiNukiyama** [comment]
@adidahiya I'm afraid of paranoid 😄🌜.

**Comment by JKillian** [comment]
Thanks @YuichiNukiyama!

**Pull request merged**